.githooks/prepare-commit-msg

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# If the branch name contains the issue number, it will append it to the
+# commit message. Example:
+#
+#   BRANCH NAME               APPEND
+#   feature/PLANET-3125-fix   PLANET-3125
+#   planet-3125               PLANET-3125
+#   some-name                 fail
+
+
+# Customize which branches should be skipped when
+# prepending commit message.
+if [ -z "$BRANCHES_TO_SKIP" ]; then
+  BRANCHES_TO_SKIP=(master develop)
+fi
+
+# Get current branch.
+BRANCH_NAME=$(git symbolic-ref --short HEAD)
+
+# Get planet4 branch.
+PLANET_BRANCH_NAME=$(echo "$BRANCH_NAME" | grep -oE "(PLANET|planet)-[0-9]*" | tr '[:lower:]' '[:upper:]')
+
+# Is current branch in excluded list?
+BRANCH_EXCLUDED=$(printf "%s\\n" "${BRANCHES_TO_SKIP[@]}" | grep -c "^$BRANCH_NAME$")
+
+# Check if planet4 branch exists in commit message.
+ISSUE_IN_COMMIT=$(grep -cE "(PLANET|planet)-[0-9]*" "$1")
+
+# If current branch is in the excluded list, nothing to do here, exit successfully.
+if [[ $BRANCH_EXCLUDED -eq 1 ]]; then
```

Comment:
Test exit codes directly instead of assigning them to a variable that isn't used elsewhere, ie removing line 25 and replacing this conditional with:

```
if printf "%s\\n" "${BRANCHES_TO_SKIP[@]}" | grep -q "^$BRANCH_NAME$"
then
  exit 0
fi
```

Tests in `if` statements don't cause set -e to exit the script.  grep returns 0 (success) if it matches and 1 if it fails, so you don't need to count number of hits just use the return code.

Then we can turn on `set -eo pipefail` https://explainshell.com/explain?cmd=set+-euxo%20pipefail with confidence.

Reply:
Could (and should) also add `set -u`, but that introduces another syntax change which I'll leave as an exercise for the intrepid.

Reply:
I used variables to make the code more readable, it gives a hint to whoever reading it about the pipeline.

Replaced it with awk.

Reply:
The number of hits is irrelevant, the test is only for whether the string is present at all, and this is where using the exit code from `grep -q` is the correct tool. Adding more complexity with awk is unwarranted.

https://github.com/progrium/bashstyle/blob/master/README.md#best-practices-and-tips
"Skip [[ ... ]] in your if-expression if you can test for exit code instead."

I don't think it improves readability more than a comment, and adding variables that are only used as a test once is an anti-pattern.